roles/python-app/tasks/main.yml

```diff
@@ -36,14 +26,29 @@
     state: "{{ item.state | default(omit) }}"
   with_items: "{{ python_app_pipdeps }}"
 
-- name: Install application
-  pip:
-    extra_args: -U
-    name: "{{ python_app_source_path }}/{{ name }}"
-    virtualenv: "{{ python_app_venv_path }}/{{ name }}"
-  when: python_app_git_checkout.changed
-
-- name: Set python-app facts
+- name: Set python-app venv fact
   set_fact: >
     {{ name | replace('-', '_') }}_venv_path={{ python_app_venv_path }}/{{ name }}
-    {{ name | replace('-', '_') }}_source_path={{ python_app_source_path }}/{{ name }}
+
+- when: "{{ python_app_git_repo | bool }}"
```

Comment:
I'm not sure I like putting the when before the block, but it's not technically wrong.